binding/flutter/lib/porcupine.dart

```diff
@@ -88,33 +92,71 @@ class Porcupine {
+  static Future<Porcupine> fromKeywordPaths(List<String?>? keywordPaths,
+          {String? modelPath, List<double>? sensitivities}) =>
+      Porcupine._create(
+          keywordPaths: keywordPaths,
+          modelPath: modelPath,
+          sensitivities: sensitivities);
 
   /// private static creator
-  static Future<Porcupine> _create({List<String?>? keywords, List<String?>? keywordPaths, String? modelPath, List<double>? sensitivities}) async {
+  static Future<Porcupine> _create(
+      {List<String?>? keywords,
+      List<String?>? keywordPaths,
+      String? modelPath,
+      List<double>? sensitivities}) async {
     if (!_resourcesExtracted) {
       await _extractPorcupineResources();
       _resourcesExtracted = true;
     }
 
-    modelPath ??= _defaultModelPath;
+    if (modelPath == null || modelPath == "") {
+      modelPath = _defaultModelPath;
+    }
+
+    if (!await File(modelPath!).exists()) {
+      try {
+        modelPath = await _extractResource(modelPath);
+      } catch (_) {
+        throw new PvArgumentError(
+            "Could not find model file at path '$modelPath'. If this is a packaged asset, ensure you have added it to your pubspec.yaml file.");
+      }
+    }
 
     if (keywords != null && keywordPaths != null) {
-      throw new PvArgumentError("Both 'keywords' and 'keywordPaths' were set. Only one of the two arguments may be set for initialization.");
+      throw new PvArgumentError(
+          "Both 'keywords' and 'keywordPaths' were set. Only one of the two arguments may be set for initialization.");
     }
 
     if (keywordPaths == null || keywordPaths.length == 0) {
       if (keywords == null || keywords.length == 0) {
-        throw new PvArgumentError("No built-in or custom keywords provided. Either 'keywords' or 'keywordPaths' must be set.");
+        throw new PvArgumentError(
+            "No built-in or custom keywords provided. Either 'keywords' or 'keywordPaths' must be set.");
       }
 
       keywordPaths = List.filled(keywords.length, "");
       for (var i = 0; i < keywords.length; i++) {
         if (BUILT_IN_KEYWORDS.contains(keywords[i])) {
           keywordPaths[i] = _builtInKeywordPaths?[keywords[i]];
         } else {
-          throw new PvArgumentError("One or more keywords are not available by default. Available default keywords are: $BUILT_IN_KEYWORDS");
+          throw new PvArgumentError(
+              "One or more keywords are not available by default. Available default keywords are: $BUILT_IN_KEYWORDS");
+        }
+      }
+    }
+
+    for (var i = 0; i < keywordPaths.length; i++) {
+      if (keywordPaths[i] == null || keywordPaths[i] == "") {
+        throw new PvArgumentError(
+            "One of the provided keyword paths was empty.");
+      }
+
+      if (!await File(keywordPaths[i]!).exists()) {
+        try {
+          keywordPaths[i] = await _extractResource(keywordPaths[i]!);
+        } catch (_) {
+          throw new PvArgumentError(
```

Comment:
does dart have a concept of nested exceptions? there is value in not throwing away whatever the original exception was, even if we're translating it upstream to a "Pv***Error"

Reply:
Not that I can find. I can nest it as part of the message parameter of the PvArgumentError though. Mostly this call just serves to check that the file exists in the app bundle, which you can't do directly.